modules/derivation/shared/src/main/scala/io/circe/derivation/Discriminator.scala

```diff
@@ -0,0 +1,12 @@
+package io.circe.derivation
+
+//Used to manage different ways of ADT serialization
+
+sealed trait Discriminator
+
+object Discriminator {
+  final case class Embedded(fieldName: String) extends Discriminator
+  final case object TypeDiscriminator extends Discriminator
+
+  lazy val default = Embedded("type")
```

Comment:
Nitpicking again, but can you put a type annotation on this, both for documentation and so it's not inferred to be `Embedded`?

Reply:
Also for something like this I think the overhead of `lazy val` probably outweighs any benefits of deferring the instantiation.

Reply:
I used the lazy val to be sure to no have concurrency problem in initialization of the value